crates/core/src/config.rs

```diff
@@ -152,3 +160,112 @@ impl SpacetimeDbFiles for FilesGlobal {
         paths::config_path()
     }
 }
+
+/// Enumeration of options for reading configuration settings.
+#[derive(Debug, PartialEq, Eq, Hash)]
+pub enum ReadConfigOption {
+    SlowQueryThreshold,
+    SlowIncrementalUpdatesThreshold,
+    SlowSubscriptionsThreshold,
+}
+
+impl ReadConfigOption {
+    pub fn type_of(&self) -> AlgebraicType {
+        AlgebraicType::U64
+    }
+}
+
+impl Display for ReadConfigOption {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let value = match self {
+            ReadConfigOption::SlowQueryThreshold => "slow_ad_hoc_query_threshold",
+            ReadConfigOption::SlowIncrementalUpdatesThreshold => "slow_incremental_query_threshold",
+            ReadConfigOption::SlowSubscriptionsThreshold => "slow_subscription_query_threshold",
```

Comment:
Final naming suggestion :)

```suggestion
            ReadConfigOption::SlowQueryThreshold => "slow_ad_hoc_query_ms",
            ReadConfigOption::SlowIncrementalUpdatesThreshold => "slow_tx_update_ms",
            ReadConfigOption::SlowSubscriptionsThreshold => "slow_subscription_query_ms",
```